sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -294,6 +294,34 @@ def _conn_key(self, db_key, db_name=None, key_prefix=None):
             key_prefix = ""
         return '{}{}:{}:{}:{}:{}:{}'.format(key_prefix, dsn, host, username, password, database, driver)
 
+    def _parse_connection_string(self, cs):
```

Comment:
Is there any method of escaping in connection strings we need to worry about? Can a password contain `;` or `=` for instance?

Reply:
Looks like it would be an issue and causes some issues with users: https://stackoverflow.com/questions/23014180/sql-server-2008-password-ending-in-a-semicolon

Reply:
FWIW -- +1, ran into this issue my self. SQL Server ODBC's docs recommend to not use special characters

> Because of connection string and initialization file grammar, keywords and attribute values that contain the characters []{}(),;?*=!@ not enclosed with braces should be avoided.

Source: https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqldriverconnect-function?redirectedfrom=MSDN&view=sql-server-ver15

Reply:
Yeah good call, technically SQL Server supports this escape method https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver15#escaping-values-in-the-connection-url 

Wrapping in curly braces. There's an example 

```
jdbc:sqlserver://localhost;username=MyUsername;password={pass";{}}word};
```

Reply:
Updated the parsing to handle escape sequences correctly